### PR TITLE
office

### PR DESCRIPTION
**Pull request by KhusanKholmatov**
created office entities, officeCreateDto, officeResponse and others

### CODE REVIEW

**Review by Sigest — APPROVED** [review]

File: src/main/java/com/exadel/coolDesking/office/OfficeRepository.java [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
 import org.springframework.stereotype.Repository;
 
+import javax.validation.constraints.NotBlank;
 import java.util.UUID;
 
 @Repository
 public interface OfficeRepository extends JpaRepository<Office, UUID> {
-}
+
+    boolean existsByName(@NotBlank @Length(min = 5, max = 64) String name);
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/office/OfficeRepository.java:13]: Constraints doesn't work here

File: src/main/java/com/exadel/coolDesking/office/OfficeRepository.java [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+
+    Office getOfficeById(UUID id);
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/office/OfficeRepository.java:15]: JpaRepository already has getById method, but better to use CrudRepository's method findById

File: src/main/java/com/exadel/coolDesking/floorPlan/FloorPlanRepository.java [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
 import java.util.UUID;
 
 
 @Repository
 public interface FloorPlanRepository extends JpaRepository<FloorPlan, UUID> {
+
+    @Query(value = "select CAST(id as varchar) from floor_plan f where f.office_id=?", nativeQuery = true)
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/floorPlan/FloorPlanRepository.java:14]: Is this CAST(id as varchar) required?

Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/floorPlan/FloorPlanRepository.java:14]: Can we use not native query?

File: src/main/java/com/exadel/coolDesking/office/OfficeService.java [hunk 4]
```diff
@@ -1,21 +1,54 @@
...
+        return officeResponse;
 
-    public Office create(Office office) {
-        return null;
     }
 
-    public Office getByid(UUID id) {
-        return null;
+    public OfficeResponse getOfficeById(UUID id) {
+        Office officeById = officeRepository.getOfficeById(id);
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/office/OfficeService.java:41]: use findById

File: src/main/java/com/exadel/coolDesking/office/OfficeController.java [hunk 5]
```diff
@@ -5,21 +5,39 @@
...
+    public ResponseEntity<?> getById(@PathVariable UUID id) {
+        OfficeResponse officeResponse = officeService.getOfficeById(id);
+        if(officeResponse.isSuccess()){
+            return ResponseEntity.status(200).body(new ArrayList(Arrays.asList(
+                    officeResponse.getOffice(),
+                    officeResponse.getMapIds()
+            )));
+        }
+
+        return ResponseEntity.status(404).body(officeResponse.getMessage());
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/office/OfficeController.java:30]: Don't do so.

1. Create exception for this situation and throw it from service layer.
2. Implement Simple exception handling 
https://www.baeldung.com/exception-handling-for-rest-with-spring#exceptionhandler

File: src/main/java/com/exadel/coolDesking/floorPlan/FloorPlanRepository.java (hunk 3 above)
Comment (KhusanKholmatov) [reply src/main/java/com/exadel/coolDesking/floorPlan/FloorPlanRepository.java:14]: yes, JDBC couldn't get UUID, so I had to cast it

File: src/main/java/com/exadel/coolDesking/office/OfficeResponseMessage.java [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package com.exadel.coolDesking.office;
+
+public interface OfficeResponseMessage {
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/office/OfficeResponseMessage.java:3]: There is one reason, why you don't need to use interface as a container for constants.
Someone can implement your interface in different, logically not connected classes.
Better to use final class with private constructor.
https://www.baeldung.com/java-constants-good-practices

File: src/main/java/com/exadel/coolDesking/office/OfficeController.java [hunk 7]
```diff
@@ -5,21 +5,39 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 import java.util.UUID;
 
 @RequiredArgsConstructor
 @RestController
 @RequestMapping(path = "/office")
 public class OfficeController {
+    private final OfficeRepository officeRepository;
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/office/OfficeController.java:17]: No repository in a controller

File: src/main/java/com/exadel/coolDesking/common/exception/ErrorAdvice.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        return new ResponseError(404, exception.getMessage(), request.getServletPath());
+    }
+
+    @ExceptionHandler(ConflictException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ResponseError handleConflictException(ConflictException exception, HttpServletRequest request){
+        return new ResponseError(400, exception.getMessage(), request.getServletPath());
+    }
+
+}
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/coolDesking/common/exception/ErrorAdvice.java:25]: this should be in a separate Pull Request
